ui/util/policy.js

```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types';
+
+/**
+ * The Policy class wraps the policy structure returned by the API
+ * and adds a number of convenience methods to it.  It's intended to be
+ * immutable; no state is stored in it, and calling JSON.stringify() on
+ * it yields the same policy structure it was initialized with.
+ */
+
+export default class Policy {
+  constructor(policy) {
+    Object.assign(this, policy);
+    Object.freeze(this);
+  }
+
+  hasDocument() {
+    return this.omb_policy_id && this.has_docnode;
+  }
+
+  getDocumentLinkProps() {
+    return {
+      route: 'document',
+      params: {
+        policyId: this.omb_policy_id,
+      },
+    };
+  }
```

Comment:
These methods *could* be computed properties, but I felt that making them explicit methods might make it easier for readers to distinguish between what are original properties of policy structs vs. extra business logic that we add. Let me know if you disagree though, we can always change them.

Reply:
I'd actually lean heavier on the properties side; when I see a method call, I assume work is being done. You raise a good point about distinguishing between API values and new logic. Maybe we make the decision about whether or not to duplicate the API's interface and, if not, keep these as methods?